### PR TITLE
Refactor robot environments into a BaseEnv class

### PR DESCRIPTION
**Pull request by ll7**
Introduce a BaseEnv class to streamline robot environment initialization and enhance visualization and recording capabilities. Implement specific robot environments inheriting from BaseEnv, along with improved configuration settings and unit tests for validation.

## Summary by CodeRabbit

Based on the comprehensive summary of changes, here are the updated release notes:

- **New Features**
	- Introduced a new base environment framework for robotic simulations.
	- Added support for robot and pedestrian environment configurations.
	- Implemented advanced navigation and route tracking capabilities.
	- Enhanced Lidar sensor and obstacle detection functionality.
	- Added functionality for saving recorded simulation states and video.

- **Improvements**
	- Developed a robust testing suite for environment components.
	- Improved metrics tracking for routes and collisions.
	- Streamlined environment configuration and initialization.

- **Testing**
	- Comprehensive test coverage for robot motion, sensor detection, and navigation.
	- Added unit tests for environment interactions and edge cases.
	- Validated pedestrian and robot movement scenarios.

- **Infrastructure**
	- Refactored import statements and logging mechanisms.
	- Improved environment settings and configuration management.

These release notes highlight the key enhancements to the robotic simulation framework, focusing on the user-visible improvements and capabilities added in this update.

### CODE REVIEW

File: tests/test_new_base_env/env_test.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+def test_can_simulate_with_pedestrians():
+    total_steps = 1000
+    env = RobotEnvFromBase()
+    env.reset()
+    for _ in range(total_steps):
+        rand_action = env.action_space.sample()
+        _, _, terminated, truncated, _ = env.step(rand_action)
+        done = terminated or truncated
+        if done:
+            env.reset()
```
[the remark this answers is not in the document]
Comment (ll7) [reply tests/test_new_base_env/env_test.py:36]: not relevant here
